src/main/java/com/internet/shop/dao/impl/ShoppingCartDaoImpl.java

```diff
@@ -44,6 +43,20 @@ public ShoppingCart update(ShoppingCart shoppingCart) {
                 .findFirst();
     }
 
+    @Override
+    public ShoppingCart addProduct(ShoppingCart cart, Product product) {
```

Comment:
it's not good to have classes with empty methods. 
In our case the easiest way would be to have only one impl for dao layer (jdbc) and remove your storage impl.
Of course if you want to have two different implementations you should write working methods then